src/components/diagrams/singleLineDiagram/single-line-diagram-content.js

```diff
@@ -0,0 +1,408 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import React, {
+    forwardRef,
+    useImperativeHandle,
+    useCallback,
+    useState,
+    useLayoutEffect,
+    useRef,
+    useEffect,
+} from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { RunningStatus } from '../../util/running-status';
+import { equipments } from '../../network/network-equipments';
+import {
+    getEquipmentTypeFromFeederType,
+    MIN_HEIGHT,
+    MIN_WIDTH,
+    MAX_HEIGHT_SUBSTATION,
+    MAX_HEIGHT_VOLTAGE_LEVEL,
+    MAX_WIDTH_SUBSTATION,
+    MAX_WIDTH_VOLTAGE_LEVEL,
+    NoSvg,
+    SvgType,
+    useDiagram,
+    useDiagramStyles,
+} from '../diagram-common';
+import withEquipmentMenu from '../../menus/equipment-menu';
+import BaseEquipmentMenu from '../../menus/base-equipment-menu';
+import withBranchMenu from '../../menus/branch-menu';
+import { SingleLineDiagramViewer } from '@powsybl/diagram-viewer';
+import {
+    isNodeInNotificationList,
+    isNodeReadOnly,
+} from '../../graph/util/model-functions';
+import { useIsAnyNodeBuilding } from '../../util/is-any-node-building-hook';
+import { fetchSvg, updateSwitchState } from '../../../utils/rest-api';
+import Alert from '@mui/material/Alert';
+import { useTheme } from '@mui/material/styles';
+import { useIntlRef, useSnackMessage } from '@gridsuite/commons-ui';
+import Box from '@mui/material/Box';
+import LinearProgress from '@mui/material/LinearProgress';
+
+const SingleLineDiagramContent = forwardRef((props, ref) => {
+    const [svg, setSvg] = useState(NoSvg);
+    const classes = useDiagramStyles();
+    const theme = useTheme();
+    const MenuBranch = withBranchMenu(BaseEquipmentMenu);
+    const network = useSelector((state) => state.network);
+    const svgRef = useRef();
+    const diagramViewerRef = useRef();
+    const { snackError } = useSnackMessage();
+    const intlRef = useIntlRef();
+    const [forceState, updateState] = useState(false);
+    const currentNode = useSelector((state) => state.currentTreeNode);
+    const [modificationInProgress, setModificationInProgress] = useState(false);
```

Comment:
There's a bug when you click on a switch. The modificationInProgress state is not set to false when the treatment is finished

Reply:
False positive. No bug